nibabel/streamlines/tck.py

```diff
@@ -0,0 +1,450 @@
+        ----------
+        fileobj : file-like object
+            An open binary file object pointing to TCK file (and ready to
+            write at the beginning of the TCK header).
+        """
+        # Fields to exclude
+        exclude = [Field.MAGIC_NUMBER,  # Handled separately.
+                   Field.NB_STREAMLINES,  # Handled separately.
+                   Field.ENDIANNESS,  # Handled separately.
+                   Field.VOXEL_TO_RASMM,  # Streamlines are always in RAS+ mm.
+                   "count", "datatype", "file"]  # Fields being replaced.
+
+        lines = []
+        lines.append(header[Field.MAGIC_NUMBER])
+        lines.append("count: {0:010}".format(header[Field.NB_STREAMLINES]))
+        lines.append("datatype: Float32LE")  # Always Float32LE.
+        lines.extend(["{0}: {1}".format(k, v)
+                      for k, v in header.items()
+                      if k not in exclude and not k.startswith("_")])
+        lines.append("file: . ")  # Manually add this last field.
+        out = "\n".join(lines)
+
+        # Check the header is well formatted.
+        if out.count("\n") > len(lines) - 1:  # \n only allowed between lines.
+            msg = "Key-value pairs cannot contain '\\n':\n{}".format(out)
+            raise HeaderError(msg)
+
+        if out.count(":") > len(lines):  # : only one per line.
+            msg = "Key-value pairs cannot contain ':':\n{}".format(out)
+            raise HeaderError(msg)
+
+        # Write header to file.
+        fileobj.write(asbytes(out))
+
+        hdr_len_no_offset = len(out) + 5
+        # Need to add number of bytes to store offset as decimal string. We
+        # start with estimate without string, then update if the
+        # offset-as-decimal-string got longer after adding length of the
+        # offset string.
+        new_offset = -1
+        old_offset = hdr_len_no_offset
+        while new_offset != old_offset:
+            old_offset = new_offset
+            new_offset = hdr_len_no_offset + len(str(old_offset))
+
+        fileobj.write(asbytes(str(new_offset) + "\n"))
+        fileobj.write(asbytes("END\n"))
+
+    @staticmethod
+    def _read_header(fileobj):
+        """ Reads a TCK header from a file.
+
+        Parameters
+        ----------
+        fileobj : string or file-like object
+            If string, a filename; otherwise an open file-like object
+            pointing to TCK file (and ready to read from the beginning
+            of the TCK header). Note that calling this function
+            does not change the file position.
+
+        Returns
+        -------
+        header : dict
+            Metadata associated with this tractogram file.
+        """
+        # Record start position if this is a file-like object
+        start_position = fileobj.tell() if hasattr(fileobj, 'tell') else None
+
+        with Opener(fileobj) as f:
+            # Read magic number
+            magic_number = asstr(f.fobj.readline()).strip()
```

Comment:
I think it is because everything is in string, here:
```
        lines = []
        lines.append(header[Field.MAGIC_NUMBER])
        lines.append("count: {0:010}".format(header[Field.NB_STREAMLINES]))
        lines.append("datatype: Float32LE")  # Always Float32LE.
        lines.extend(["{0}: {1}".format(k, v)
                      for k, v in header.items()
                      if k not in exclude and not k.startswith("_")])
        lines.append("file: . ")  # Manually add this last field.
        out = b"\n".join(lines)
```